core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CloudflowFlinkPlugin.scala

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cloudflow.sbt
+
+import java.nio.charset.StandardCharsets._
+import java.util.Base64
+import java.util.zip.Deflater
+import java.io.ByteArrayOutputStream
+
+import com.typesafe.config._
+import sbt._
+import sbt.Keys._
+import sbtdocker._
+import sbtdocker.DockerKeys._
+import com.typesafe.sbt.packager.Keys._
+import com.typesafe.sbt.packager.archetypes._
+import spray.json._
+import JsonUtils._
+import cloudflow.sbt.CloudflowKeys._
+import cloudflow.blueprint._
+import cloudflow.blueprint.StreamletDescriptorFormat._
+import cloudflow.sbt.CloudflowKeys._
+import cloudflow.blueprint.StreamletDescriptor
+
+object CloudflowFlinkPlugin extends AutoPlugin {
+  val AppHome = "${app_home}"
+  val AppTargetDir: String = "/app"
+  val appTargetSubdir: String ⇒ String = dir ⇒ s"$AppTargetDir/$dir"
+  val AppJarsDir: String = "app-jars"
+  val DepJarsDir: String = "dep-jars"
+  val optAppDir = "/opt/cloudflow/"
+  final val flinkVersion = "1.9.2"
+  final val scalaVersion = "2.12"
+  final val CloudflowFlinkDockerBaseImage = s"lightbend/flink:cloudflow-flink-$flinkVersion-scala-$scalaVersion"
```

Comment:
I am working on the spark part and I think we need to add a cloudflow specific version to it because these files may change from one cloudflow version to the other while Flink may not: ```flink:cloudflow-flink-$flinkVersion-scala-$scalaVersion-1.3.1```

Reply:
cloudflowVersion then comes from BuildInfo? We can't hard code that in any case?
I think I would prefer it without a cloudflow specific version.. (the base image might not change between cloudflow versions)

Reply:
@RayRoestenburg or it could change, so to be on the safe I suggest that as we have done in the past anyway(if you check our current published images they have a cloudflow release version as part of their tag). During development we need to reference something valid which we can delete later.

Reply:
@skonto I thought that since we r building a Flink image, why should it include a cloudflow version ? This will be hosted in dockerhub and could be used by other projects as well - right ?